### PR TITLE
Periodic tasks refactor

### PR DESCRIPTION
**Pull request by xetorthio**
Once every second the session run a list of periodic tasks on every instance concurrently. We use these tasks to do things like:
- Collect mem and cpu stats
- Check if instance is part of a swarm cluster
- Broadcast information to connected clients

### CODE REVIEW

File: services/docker.go [hunk 1]
```diff
@@ -43,20 +40,8 @@ func GetContainerInfo(id string) (types.ContainerJSON, error) {
...
-	cli := &http.Client{
-		Transport: transport,
-	}
-	c, err := client.NewClient(host, client.DefaultVersion, cli, nil)
-	if err != nil {
-		return types.Info{}, err
-	}
-	return c.Info(context.Background())
+func GetDaemonInfo(i *Instance) (types.Info, error) {
+	return i.dockerClient.Info(context.Background())
```
Comment (marcosnils) [inline services/docker.go]: dockerClient couldn't be `nil` here if it couldn't connect to the daemon?. Shoudln't we check that?

File: services/session.go [hunk 2]
```diff
@@ -48,6 +54,49 @@ func (s *Session) AddNewClient(c *Client) {
...
+						log.Println("Could not connect to DinD docker daemon", err)
+					} else {
+						i.dockerClient = c
+					}
+				}
+				go func() {
+					for _, t := range periodicTasks {
+						t.Run(i)
+					}
+					wg.Done()
```
Comment (marcosnils) [inline services/session.go]: Isn't better to defer wg.Done()? if a panic is raised then Done() won't be called.

File: services/session.go [hunk 3]
```diff
@@ -48,6 +54,49 @@ func (s *Session) AddNewClient(c *Client) {
...
+					}
+					c, err := client.NewClient(fmt.Sprintf("http://%s:2375", i.IP), client.DefaultVersion, cli, nil)
+					if err != nil {
+						log.Println("Could not connect to DinD docker daemon", err)
+					} else {
+						i.dockerClient = c
+					}
+				}
+				go func() {
+					for _, t := range periodicTasks {
```
Comment (marcosnils) [inline services/session.go]: periodicTasks could have a `RunAll` method that runs all the tasks that it has instead of having that ugly loop here. WDYT?

Comment (xetorthio) [inline services/session.go]: I don't think it makes the code any better or easier to read. This for loop is only 1 line. Having it here or in the `task.go` in a new func is not going to change anything.

File: services/session.go (hunk 2 above)
Comment (xetorthio) [inline services/session.go]: fixed

File: services/session.go [hunk 4]
```diff
@@ -89,7 +89,7 @@ func (s *Session) SchedulePeriodicTasks() {
 					for _, t := range periodicTasks {
 						t.Run(i)
 					}
-					wg.Done()
+					defer wg.Done()
```
Comment (marcosnils) [inline services/session.go]: needs to be before the `Run` call, otherwise it won't be registered :confused:

**Pull request merged**